llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4240,27 +4235,12 @@ ElementCount LoopVectorizationCostModel::getMaximizedVFForTarget(
         ComputeScalableMaxVF);
     MaxVectorElementCountMaxBW = MinVF(MaxVectorElementCountMaxBW, MaxSafeVF);
 
-    // Collect all viable vectorization factors larger than the default MaxVF
-    // (i.e. MaxVectorElementCount).
-    SmallVector<ElementCount, 8> VFs;
+    // Set the max VF to the largest viable vectorization factor less than or
+    // equal to the max vector element count.
     for (ElementCount VS = MaxVectorElementCount * 2;
          ElementCount::isKnownLE(VS, MaxVectorElementCountMaxBW); VS *= 2)
-      VFs.push_back(VS);
-
-    // For each VF calculate its register usage.
-    auto RUs = calculateRegisterUsage(VFs);
+      MaxVF = VS;
```

Comment:
Given these element counts are powers of 2 (it appears `bit_floor` is used above), I think this is just:
```
MaxVF = MaxVectorElementCountMaxBW;
```
(assuming `MaxVectorElementCount <= MaxVectorElementCountMaxBW`)

Reply:
I added an assertion checking if `MaxVectorElementCount <= MaxVectorElementCountMaxBW` and an AArch64 test failed (Transforms/LoopVectorize/AArch64/smallest-and-widest-types.ll) so that condition doesn't always hold. So I've replaced the loop with a check and an assignment.